src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -90,21 +90,29 @@ void SerialCouplingScheme::receiveAndSetTimeWindowSize()
 
 void SerialCouplingScheme::exchangeInitialData()
 {
+  bool initialDataExchange = true;
+
   // F: send, receive, S: receive, send
   if (doesFirstStep()) {
     if (receivesInitializedData()) {
-      receiveData(getM2N(), getReceiveData());
+      receiveData(getM2N(), getReceiveData(), initialDataExchange);
       checkDataHasBeenReceived();
     } else {
       initializeWithZeroInitialData(getReceiveData());
     }
+    if (sendsInitializedData()) { // this send/recv pair is only needed, if no substeps are exchanged.
+      sendData(getM2N(), getSendData(), initialDataExchange);
+    }
```

Comment:
I don't like this additional send/recv pair a lot, but I think it is the easiest way to fix the bug for `substeps="false"`. Comes with some overhead for `substeps="true"` during initialization, but as mentioned above: We might also benefit from this, if we generally don't communicate data for `WINDOW_START` if `initialDataExchange = false`.